### PR TITLE
Commitment Discount Appendix

### PR DESCRIPTION
**Pull request by cnharris10**
An appendix detailing how various commitment discount purchases and usage records are modeled through various examples.

### CODE REVIEW

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:11]: This needs to be genericized as `partial` or something similar, so as to include partial upfront that isn't exactly half.

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:3]: Would love this definition to not include the word `discount` in the first line. Something like:

```suggestion
A commitment discount is a rate reduction on preselected SKUs in exchange for an obligated usage or spend amount over a predefined term. Commitment discounts typically consist of a set of purchase and usage records within cost and usage datasets.
```

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:13]: This is the first mention of the term `spend-based`.  Given that this page is the explanation of a commitment discount, it should be written with the assumption that the reader brings no previous understanding with them.  This term deserves a definition earlier if you are going to use it here.

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:10]: Is `Monthly` the right term here?  There may be other grains we want to include here.  Suggest using `Recurring` or similar instead.

File: specification/appendix/commitment_discounts.md
Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:3]: I'm going to keep this as is to maintain consistency with this pr, the new `CommitmentDiscunt*` columns, and the FinOps terminology page, so actually need to add the missing word, "model".

image

File: specification/appendix/commitment_discounts.md
Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:10]: Recurring works.

File: specification/appendix/commitment_discounts.md
Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:13]: I think by that logic, we would also need add definitions for:
- billing period
- charge period
- amortization
- term
...

Other than redefining `commitment discount` in the first sentence, I would rather link the glossary to the appendix rather than regurgitate definitions in 2 places. 

Thoughts?

File: specification/markdownlnt.cfg
Comment (cnharris10) [inline specification/markdownlnt.cfg:5]: This pr will not pass markdownlinter without this change.  When `spec.md` is created, an `<a>` tag is added to the end of the file causing the linter to fail.

image

**Comment by jpradocueva** [comment]
### Action Items from Maintainers' call on Aug 26:
- [ ] [Maintainers-[#528](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/528)] Chris - @cnharris10 : To review and discuss with relevant stakeholders (Zach - @AWS-ZachErdman , Michael - @flanakin , Tim - @timwright2000 ) and incorporate any feedback.

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:36]: Would prefer to add consistency in one of the following ways:

```suggestion
Recurring purchases are allocated across all corresponding charge periods of the term when `ChargeCategory` is "Purchase", `ChargeFrequency` is "One-Time", and `CommitmentDiscountPurchasedQuantity` and `CommitmentDiscountUnit` are reflected only for that charge period.
```

```suggestion
Recurring purchases are allocated across all corresponding charge periods of the term when the charge category (`ChargeCategory`) is "Purchase", charge frequency (`ChargeFrequency`) is "One-Time", and the commitment discount's purchase quantity (`CommitmentDiscountPurchasedQuantity`) and units (`CommitmentDiscountUnit`) are reflected only for that charge period.
```

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:133]: This section should remind the reader that there's only one resource in play in the example.

File: specification/appendix/commitment_discounts.md
Comment (salpaysenturus) [inline specification/appendix/commitment_discounts.md:5]: While I am not willing to hold up the approval of this PR, I do think that this document will strongly benefit from either 1) presenting an example of how size flexibility surfaces in the data, or 2) explicitly stating that a deeper explanation of size flexibility is beyond the scope of this document.

**Review by salpaysenturus — APPROVED** [review]
Left comments, but approved.  Good job, my friend.

File: specification/appendix/commitment_discounts.md
Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:5]: Agree, would like to iterate to that goal too.

Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:5]: Size Flexibility Appendix: https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/535

File: specification/appendix/commitment_discounts.md
Comment (ijurica) [inline specification/appendix/commitment_discounts.md:36]: Suggestion based on Scenario 2: Recurring

```suggestion
Recurring purchases are allocated across all corresponding charge periods of the term when `ChargeCategory` is "Purchase", `ChargeFrequency` is "Recurring", and `CommitmentDiscountPurchasedQuantity` and `CommitmentDiscountUnit` are reflected only for that charge period.
```

Comment (ijurica) [reply specification/appendix/commitment_discounts.md]: Regarding the editorial guidelines, which format are we supposed to use for column IDs — for instance, `ChargeFrequency` or ChargeFrequency? 
Based on https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/501, I assume it's the latter.

Comment (cnharris10) [reply specification/appendix/commitment_discounts.md]: Will incorporate styling guidelines across all pr's after merging content.

**Review by ahullah — APPROVED** [review]

**Comment by jpradocueva** [comment]
### Action Items from TF-2 call on Sep 18:
- [ ] [TF2-[#428](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/528)] Chris, @cnharris10 : Update the PR to reflect the term "partial" instead of "hybrid" and submit for final review.
- [ ] [TF2-[#428](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/528)] All members: Review the updated appendix in preparation for the next meeting’s discussion.

File: specification/appendix/commitment_discounts.md
Comment (ijurica) [inline specification/appendix/commitment_discounts.md:7]: When we discussed the other two commitment utilization-related columns, we decided 

- to be as vague as possible about commitment flexibility 
- and to avoid mentioning the billing currency (in the case of spend-based commitments). 
 
For the same reason, I suggest omitting these two chapters and incorporating the mention of commitment flexibility into the first chapter.

File: specification/appendix/commitment_discounts.md
Comment (ijurica) [inline specification/appendix/commitment_discounts.md:15]: Personally, I prefer using the terms All-Upfront, No-Upfront, and Partial-Upfront for payment models to avoid potential confusion with the Recurring Charge Frequency.
As for the descriptions, I suggest we focus on mentioning One-Time and/or Recurring purchases.

File: specification/appendix/commitment_discounts.md
Comment (cnharris10) [reply specification/appendix/commitment_discounts.md:7]: Chris to rephrase general directions to point towards specific examples

**Review by ijurica — APPROVED** [review]

**Comment by jpradocueva** [comment]
Approved by the Members on the Sep 19 call.

File: specification/appendix/commitment_discounts.md
Comment (udam-f2) [inline specification/appendix/commitment_discounts.md:11]:
```suggestion
While customers are bound to the term of a commitment discount, providers offer some or all of the following payment options before or during the term:
```

File: specification/appendix/commitment_discounts.md
Comment (udam-f2) [inline specification/appendix/commitment_discounts.md:3]:
```suggestion
A commitment discount is a billing discount model that offers reduced rates on preselected SKUs in exchange for an obligated usage or spend amount over a predefined term. Commitment discounts typically consist of purchase and usage records within cost and usage datasets.
```

**Review by udam-f2 — APPROVED** [review]
Nit: this looks like its missing a bunch of links to various glossary entries and columns etc. Also seems to be using other formatting patterns e.g. `CommitmentDiscountQuantity`when after the first one, we would just be using italics per the guidelines.

Fine if fixed afterwards during consistency review.

**Comment by jpradocueva** [comment]
@udam-f2, @cnharris10, I suggest that this can be fixed as part of the consistency review. I would like to issue a notice to the group with the links to the baseline for the consistency review, and for that, we need to merge this PR. Suggestions?

**Pull request merged**
